spec/forms/assessor_interface/assessment_declaration_decline_form_spec.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+RSpec.describe AssessorInterface::AssessmentDeclarationDeclineForm,
+               type: :model do
+  let(:assessment) { create(:assessment, :started) }
+  let(:session) { {} }
+  subject(:form) { described_class.new(session:, assessment:) }
+  it { is_expected.to validate_presence_of(:declaration) }
+
+  let(:assessment) { create(:assessment, :decline_with_no_failure_Reasons) }
```

Comment:
This `assessment` overwrites the `assessment` definition on line 6 meaning both exceptions are run with it. You'll probably want to define two `context` blocks for a declined assessment and a started one. Then you can also test that the form _doesn't_ validate the presence of `recommendation_assessor_note` when the assessment has failure reasons.